docs/routers/OPNsense.md

```diff
@@ -0,0 +1,98 @@
+
+Beware: Link local IPv6 are not reachable from other VLANs.
+
+
+All client devices on your network should now automatically be configured with a single IPv4 and single IPv6 address for DNS resolution. If you don't see your client devices update quickly enough, you *can* toggle Wi-Fi off/on or disconnect the Ethernet cable for a couple of seconds and then reconnect it.
+
+### Method 2: Using your Pi-hole as DNS server for your OPNsense
+
+If you'd rather have your OPNsense forward all DNS traffic to Pi-hole, you can complete this *alternative* step.
+Doing so will have DNS queries taking the following path:
+
+```
+(Clients) -> OPNsense -> Pi-hole -> Upstream DNS Server
+```
+
+!!! warning
+    Keep in mind that if you are using your OPNsense IP address as an upstream resolver for your Pi-hole together with this configuration, ***this will cause a resolver loop***. Make sure to have another upstream DNS server(s) set in your Pi-hole's configuration, or use Pi-hole as [your own recursive DNS server with Unbound](https://docs.pi-hole.net/guides/dns/unbound/?h=unbound#configure-unbound).
+
+1. In Pi-hole, navigate to `Settings -> DNS` and ensure you have at least one external upstream DNS server enabled.
+2. In OPNsense navigate to `Settings -> General -> Networking`.
+3. Under DNS Servers, enter the IPv4 and or IPv6 address of your Pi-hole host, and set the gateway to your WAN interface.
+4. Uncheck `Allow DNS server list to the overridden by DHCP/PPP on WAN`.
+5. Click Save.
+
+### Block other DNS servers (optional)
+
+Some clients have hardcoded DNS settings. To prevent them from circumventing our Pi-Hole, we can optionally block all DNS request that are not coming from our Pi-Hole.
+Warning, if your device does not have any fallback from the hardcoded DNS settings, this could break it from resolving.
+
+We first create an Alias for Pi-holes IPs.
+Go to **Firewall** --> **Aliases** scroll down and click on the plus button to create a new alias.
+Give it a Name and add the IPv4 and IPv6 into the "Content" field and then click save.
+
+![Screenshot of OPNsense Aliases](../images/routers/opnsene-alias.png)
+
+Now go to **Firewall** --> **Rules** and click on the interface that you want to modify.
+
+We create one rule to allow pihole to do DNS over IPv4 + IPv6.
+After that rule, we create a rule to reject ALL DNS. It is important that this reject rule is after the other rule.
+OPNsense does the first rule that matches, if we have our reject rule first, Pi-hole can not reach any DNS servers.
+It should look something like this:
+
+![Screenshot of OPNsense Firewall rules](../images/routers/opnsense-firewall-rule.png)
+
+We don't need a rule to allow for example device 10.0.51.10 to reach our pihole at 10.0.51.2, since they are both on the same network, DNS request will not be routed through OPNsense. Instead the client 10.0.51.10 will directly contact Pi-hole for DNS requests.
```

Comment:
```suggestion
We don't need a rule to allow for example device 10.0.51.10 to reach our Pi-hole at 10.0.51.2, since they are both on the same network, DNS request will not be routed through OPNsense. Instead the client 10.0.51.10 will directly contact Pi-hole for DNS requests.
```